src/NuGet.Core/NuGet.Commands/RestoreCommand/Utility/NoOpRestoreUtilities.cs

```diff
@@ -30,7 +30,8 @@ internal static bool IsNoOpSupported(RestoreRequest request)
         }
 
         /// <summary>
-        /// The cache file path is $(MSBuildProjectExtensionsPath)\$(project).nuget.cache
+        /// The cache file path will be $(RestoreOutputPath)\$(project).nuget.cache
```

Comment:
This is really nit, but we should prefer the `MSBuildProjectExtensionsPath` as first. It's the one that's used in PackageReference. 

The DOtnetToolReference scenario is specially, so that comment can go 2nd. 
project.json goes into the projectDirectory.

It's messy and PackageReference is really the only style most people need to think about :)

Reply:
This is not user facing doc. As such it should at least be accurate.

Still, in our build setup, we do put restore assets and targets into separate folders.

Reply:
> restore assets and targets

And by this you mean `project.assets.json` in one directory and `nuget.g.targets` in a different one?

Reply:
Yes

Reply:
I wouldn't expect that to work in VS. 
In VS we don't read RestoreOutputPath.

Reply:
I didn't know that. Let me see if I can repro it.

Reply:
For context, this is where we read it in VS (for SDK based projects) https://github.com/NuGet/NuGet.Client/blob/dev/src/NuGet.Clients/NuGet.SolutionRestoreManager/VsSolutionRestoreService.cs#L217. It's coming from: https://github.com/dotnet/project-system/blob/5a3fe3edc82a7ee5c64b044cd375782275cc9fe0/src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/Rules/NuGetRestore.xaml#L32-L34. 

In commandline it's https://github.com/NuGet/NuGet.Client/blob/0df34003a113462cfa02e369491180686a08beaa/src/NuGet.Core/NuGet.Build.Tasks/NuGet.targets#L659-L662. 

We only read RestoreOutputPath to handle tool restore, which only works commandline. 

If we had the knowledge we have today, we wold've definitely not introduced RestoreOutputPath at all, but that ship has long sailed :)